File: pset6/readability.py
# Computes the approximate grade level needed to comprehend some text.

from cs50 import get_string
import re

# Prompts user for text input.
text = get_string("Enter your text: ")

# Counting letters and words.
# Found here https://stackoverflow.com/questions/24878174/how-to-count-digits-letters-spaces-for-a-string-in-python
letters = sum(c.isalpha() for c in text)
words  = sum(c.isspace() for c in text) + 1

# Counting sentences via . ! ? chars.
sentences = text.count('.') + text.count('!') + text.count('?')

# Main grade (float) calculation using the given formula
grade = round(0.0588 * ((float(letters) / (float(words))) * 100.0) - 0.296 * ((float(sentences) / (float(words))) * 100.0) - 15.8)

if grade > 0 and grade < 17:
    print(f"Grade {grade}")
elif grade < 1:
    print ("Before Grade 1")
else:
    print("Grade 16+")

File: pset2/readability.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

// Should become a readability grading program.

int main(void)

{

    string s = get_string("Text:  "); // Prompts user for text input.

    // Counting letters.
    int letters = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {

        if (isalpha(s[i])) // A prettier version of:   if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letters++;
        }

    }

    // Counting spaces.

    int spaces = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {

        if (isspace(s[i])) // Using the library command to count spaces.
        {
            spaces++;
        }

    }

    // Counting sentences via . ! ? chars.

    int sentences = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {

        if (s[i] == '.' || s[i] == '!' || s[i] == '?') // Using normal operators to discern chars.
        {
            sentences++;
        }

    }

    // Main grade calculation
    float grade;
    grade = 0.0588 * (((float) letters / ((float) spaces + 1)) * 100) - 0.296
            * (((float) sentences / ((float) spaces + 1)) * 100) - 15.8;

    if (grade >= 1 && grade <= 16)
    {
        printf("Grade %.0f\n", round(grade)); // for normal grades.
    }

    if (grade < 1)
    {
        printf("Before Grade 1\n"); // for sub1.
    }

    if (grade > 16)
    {
        printf("Grade 16+\n"); // for 16+.
    }

}
